components/ai_chat/ai_chat_api.cc

```diff
@@ -113,40 +153,78 @@ void AIChatAPI::QueryPrompt(ResponseCallback callback,
   dict.Set("top_p", 0.999);
   dict.Set("model", model_name);
   dict.Set("stop_sequences", std::move(stop_sequences));
-  dict.Set("stream", false);
+  dict.Set("stream", true);
 
-  base::flat_map<std::string, std::string> headers;
-  headers.emplace("x-brave-key", BUILDFLAG(BRAVE_SERVICES_KEY));
+  return dict;
+}
 
-  DVLOG(1) << __func__ << " Prompt: " << prompt << "\n";
-  DVLOG(1) << __func__ << " Using model: " << model_name;
+void AIChatAPI::OnDataReceived(base::StringPiece string_piece,
+                               base::OnceClosure resume) {
+  std::vector<std::string> stream_data = base::SplitString(
```

Comment:
We should check for a success/error http response code like `APIRequestResult` can (or re-use that code). Additionally, a request could still come back with the Content-Type response header of anything, e.g. `application/json` if it's an error. We should check for `text/event-stream` before assuming.